react/ProductCustomizer.js

```diff
@@ -11,7 +11,15 @@ class ProductCustomizerIndex extends Component {
 
     if (loading) return <Spinner />
 
-    const hasSchema = head(product.items).calculatedAttachments
+    let hasSchema
+    const { calculatedAttachments } = head(product.items)
+
+    try {
+      const schema = JSON.parse(calculatedAttachments)
+      hasSchema = Object.keys(schema.properties).length > 0
```

Comment:
tem certeza que `schema` vai ter `properties`?

Reply:
@lbebber Já está dentro de um try/catch por isso, se não tiver properties no schema, que é obrigatório, não tem nada o que o app renderizar

Reply:
isn't this `hasSchema = schema && schema.properties && schema.properties.length` or `hasSchema = path(['properties', 'length'], schema`? Try/catch seems a bit harsh.

What's the point of `JSON.parse`? Is `calculatedAttachments` a string?

Reply:
calculatedAttachments is a JSON schema, and if it can't be created (i.e. attachments don't exist or are not the right type of attachment), returns an empty JSON `{}`.
Both of your examples would not work because properties is an Object, not an Array, that's why I put an explicit conversion to array. Do you know if ramda has a way to check if Object is empty?

Reply:
https://ramdajs.com/docs/#isEmpty

Reply:
@brunojdo Acabei de usar isso, nem percebi que funcionava pra objeto também! Vou atualizar pra usar isEmpty

Reply:
@brunojdo Updated the code and the workspace for testing